From b967e6831bdcc818018d0d066c2f76fccc234e5e Mon Sep 17 00:00:00 2001
From: "yi.zhang1" <yi.zhang1@amlogic.com>
Date: Tue, 21 Nov 2023 10:22:17 +0800
Subject: [PATCH 6/9] transcoding: transcoding use too many time [1/1]

PD#SWPL-146506

Problem:
transcoding use too many time to transcode.

Solution:
set sync property FALSE.

Verify:
local

Signed-off-by: yi.zhang1 <yi.zhang1@amlogic.com>
Change-Id: I8f1065ce652a0b65f45731abc0aab67aab3499a3
---
 gst-aml-videotranscoding/video_transcoding.c  | 2 +-
 videotranscoding-demo/videotranscoding_demo.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/gst-aml-videotranscoding/video_transcoding.c b/gst-aml-videotranscoding/video_transcoding.c
index f3ba15e..761a347 100644
--- a/gst-aml-videotranscoding/video_transcoding.c
+++ b/gst-aml-videotranscoding/video_transcoding.c
@@ -203,7 +203,7 @@ HANDLE video_transcoding_init( video_transcoding_param *param, int argc, char **
     g_object_set(data->video_enc,"bitrate", param->bitrate_kb, "gop", param->gop_size, NULL);
 
     /*configure the app sink */
-    g_object_set (data->app_sink, "emit-signals", TRUE, "caps", sink_caps, NULL);
+    g_object_set (data->app_sink, "emit-signals", TRUE, "caps", sink_caps, "sync", FALSE, NULL);
     g_signal_connect (data->app_sink, "eos", G_CALLBACK (send_eos_event_to_app), data);
 
     //"async", FALSE,
diff --git a/videotranscoding-demo/videotranscoding_demo.c b/videotranscoding-demo/videotranscoding_demo.c
index a1056d7..a7c8bf1 100644
--- a/videotranscoding-demo/videotranscoding_demo.c
+++ b/videotranscoding-demo/videotranscoding_demo.c
@@ -350,7 +350,7 @@ int main(int argc, char *argv[]) {
     }
 
     /* Set appsink */
-    g_object_set(G_OBJECT(data->sink), "emit-signals", TRUE, NULL);
+    g_object_set(G_OBJECT(data->sink), "emit-signals", TRUE, "sync", FALSE, NULL);
     g_signal_connect(G_OBJECT(data->sink), "new-sample", G_CALLBACK(new_sample), NULL);
 
     /* Connect the "pad-added" signal handler */
-- 
2.25.1

